internal/policy/container/has_modified_files_test.go

```diff
@@ -0,0 +1,214 @@
+	var (
+		hasModifiedFiles HasModifiedFilesCheck
+		layers           []string
+		pkgRef           map[string]packageFilesRef
+	)
+
+	BeforeEach(func() {
+		pkgRef = make(map[string]packageFilesRef)
+		pkgRef["firstlayer"] = packageFilesRef{
+			LayerFiles: []string{
+				"this",
+				"is",
+				"not",
+				"prohibited",
+			},
+			LayerPackages: map[string]struct{}{
+				"foo-1.0": {},
+				"bar-1.0": {},
+				"baz-2.0": {},
+			},
+			LayerPackageFiles: map[string]string{
+				"this": "foo-1.0",
+				"is":   "bar-1.0",
+				"not":  "baz-2.0",
+			},
+			HasRPMDB: true,
+		}
+
+		pkgRef["secondlayer"] = packageFilesRef{
+			LayerFiles: []string{
+				"there",
+				"are",
+				"no",
+				"prohibited",
+				"duplicates",
+			},
+			LayerPackages: map[string]struct{}{
+				"foo-1.0": {},
+				"bar-1.0": {},
+				"baz-2.0": {},
+				"boz-3.0": {},
+			},
+			LayerPackageFiles: map[string]string{
+				"this":  "foo-1.0",
+				"is":    "bar-1.0",
+				"not":   "baz-2.0",
+				"no":    "boz-3.0",
+				"there": "boz-3.0",
+			},
+			HasRPMDB: true,
+		}
+		pkgRef["lastlayer"] = packageFilesRef{
+			LayerFiles: []string{
+				"prohibited",
+			},
+			LayerPackages: map[string]struct{}{
+				"foo-1.0": {},
+				"bar-1.0": {},
+				"baz-2.0": {},
+				"boz-3.0": {},
+			},
+			LayerPackageFiles: map[string]string{
+				"this": "foo-1.0",
+				"is":   "bar-1.0",
+				"not":  "baz-2.0",
+			},
+		}
+		layers = []string{
+			"firstlayer",
+			"secondlayer",
+			"lastlayer",
+		}
+	})
+
+	Context("Checking if it has any modified RPM files", func() {
+		When("there are no modified RPM files found", func() {
+			It("should pass validate", func() {
+				ok, err := hasModifiedFiles.validate(context.Background(), layers, pkgRef)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ok).To(BeTrue())
+			})
+		})
+		When("there is a modified RPM file found", func() {
+			var pkgs map[string]packageFilesRef
+			BeforeEach(func() {
+				pkgs = pkgRef
+				pkgSecondLayer := pkgRef["secondlayer"]
+				pkgSecondLayer.LayerFiles = append(pkgs["secondlayer"].LayerFiles, "this")
+				pkgs["secondlayer"] = pkgSecondLayer
+			})
+			It("should not pass Validate", func() {
+				ok, err := hasModifiedFiles.validate(context.Background(), layers, pkgs)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ok).To(BeFalse())
+			})
+		})
+		When("a package is updated", func() {
+			var pkgs map[string]packageFilesRef
+			BeforeEach(func() {
+				pkgs = pkgRef
+				pkgSecondLayer := pkgs["secondlayer"]
+				pkgSecondLayerPackageFiles := pkgSecondLayer.LayerPackageFiles
+				pkgSecondLayerPackageFiles["this"] = "foo-2.0"
+				pkgSecondLayer.LayerPackageFiles = pkgSecondLayerPackageFiles
+				pkgs["secondlayer"] = pkgSecondLayer
+			})
+			It("should pass validate", func() {
+				ok, err := hasModifiedFiles.validate(context.Background(), layers, pkgs)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ok).To(BeTrue())
+			})
+		})
+		When("a package is removed", func() {
```

Comment:
Do we want a test for adding a package?

Reply:
I think we already have that?

Reply:
Maybe that's covered in the modified and update. I was thinking we might have a test where we add a 3rd layer, but if that doesn't buy anything, maybe it's not needed.

Reply:
It's covered in the main test. If you look at the data, you will see that a package is "added" between layer one and two.